onnxruntime/contrib_ops/cpu/quadric/quadric_custom_op.cc

```diff
@@ -0,0 +1,274 @@
+        TensorShape output_shape = onnxruntime::utils::GetTensorShapeFromTensorShapeProto(*graph_output_shape);
+
+        // if size < 0 we have a symbolic dimension and need to use a temporary OrtValue in the subgraph execution
+        if (output_shape.Size() < 0) {
+          symbolic_dim_in_shape = true;
+        } else {
+          auto* tensor = context_.Output(index, output_shape);
+
+          if (!tensor)
+            return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Failed to create output tensor for ", graph_output->Name());
+
+          outputs_.push_back({AllocationType::SubgraphOutput, *context_.GetOutputMLValue(index)});
+        }
+      }
+
+      if (!graph_output_shape || symbolic_dim_in_shape) {
+        // we still need a value to put in the feeds we give to the execution frame, so just use an empty MLValue
+        outputs_.push_back({AllocationType::Delayed, {}});
+      }
+    } else if (graph_output_type->has_sequence_type()) {
+      auto* seq_tensor = context_.Output<TensorSeq>(index);
+      if (!seq_tensor)
+        return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Failed to create output tensor for ", graph_output->Name());
+      outputs_.push_back({AllocationType::SubgraphOutput, *context_.GetOutputMLValue(index)});
+    } else {
+      // Shouldn't hit this as the kernel assignment logic should check for the types before assigning this kernel
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
+                             "Only tensors, tensor sequence, optional tensor, and optional tensor sequence types are supported");
+    }
+
+    ++index;
+  }
+
+  return Status::OK();
+}
+
+Status QuadricCustomOpImpl::Execute(const FeedsFetchesManager& ffm) {
+  Status status = Status::OK();
+
+  auto num_inputs = context_.InputCount();
+  std::vector<OrtValue> feeds;
+  feeds.reserve(num_inputs);
+
+  // This will contain *all* inputs, including initializers
+  for (int i = 0; i < num_inputs; ++i) {
+    feeds.push_back(*context_.GetInputMLValue(i));
+  }
+
+  std::vector<OrtValue> fetches;
+  std::unordered_map<size_t, IExecutor::CustomAllocator> fetch_allocators;
+
+  fetches.reserve(info_.num_outputs);
+  for (int i = 0; i < info_.num_outputs; ++i) {
+    fetches.push_back(outputs_[i].second);
+
+    if (outputs_[i].first == AllocationType::Delayed) {
+      // functor to forward the allocation request from the subgraph to the If node's context so that the
+      // allocation plan for the If node's output is used.
+      fetch_allocators[i] = [this, i, &fetches](const TensorShape& shape, const OrtDevice& location,
+                                                OrtValue& ort_value, bool& allocated) {
+        // if the device the If output is allocated on does not match the required device for the subgraph output
+        // we don't update the provided OrtValue and return false for 'allocated'.
+        // the execution frame will allocate a buffer on the required device, and the fetches copy
```

Comment:
`fetches copy` seems like an error but not sure what you intended to write here

Reply:
I copied this verbatim from `If` node handling. So I'm also not positive about the intention.